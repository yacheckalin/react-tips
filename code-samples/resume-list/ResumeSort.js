import React, { useState, useContext } from "react";
import PropTypes from "prop-types";
import ResumeListContext from "./ResumeListContext";

import { mapSortByToDefaultValue, mapStateToSortObject } from "./helpers";

const ResumeSort = ({ onSort }) => {
  document.addEventListener("DOMContentLoaded", function () {
    var elems = document.querySelectorAll("select");
    M.FormSelect.init(elems, {});
  });

  const context = useContext(ResumeListContext);

  const { sortBy, sortDirection } = context;
  const [defaultValue, setDefaultValue] = useState(
    mapSortByToDefaultValue(sortBy, sortDirection)
  );

  return (
    <div className="input-field col s3 ">
      <select
        defaultValue={defaultValue}
        onChange={(e) => {
          onSort(mapStateToSortObject(e.target.value));
        }}
      >
        <option value="1">Priority &#x2191;</option>
        <option value="2">Priority &#x2193;</option>
      </select>
      <label>Sort By:</label>
    </div>
  );
};

ResumeSort.propTypes = {
  onSort: PropTypes.func.isRequired,
};

export default ResumeSort;
