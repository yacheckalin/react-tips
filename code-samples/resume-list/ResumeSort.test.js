import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";

import ResumeSort from "./ResumeSort";

describe("<ResumeSort />", () => {
  const props = {
    onSort: jest.fn(),
  };
  test("renders and displays properly", () => {
    const wrapper = shallow(<ResumeSort {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
