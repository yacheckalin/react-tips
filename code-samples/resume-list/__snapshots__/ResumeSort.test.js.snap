// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResumeSort /> renders and displays properly 1`] = `
<div
  className="input-field col s2 "
>
  <select
    defaultValue={1}
    onChange={[Function]}
  >
    <option
      value="1"
    >
      Priority ↑
    </option>
    <option
      value="2"
    >
      Priority ↓
    </option>
  </select>
  <label>
    Sort By:
  </label>
</div>
`;
